io/disk/Avago_storage_adapter/avago9361_vd.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+#
+# Copyright: 2016 IBM
+# Author: Venkat Rao B <vrbagal1@linux.vnet.ibm.com>
+
+
+"""
+This scripts performs Virtual Drive(VD) operations on drives
+"""
+
+import time
+from avocado import Test
+from avocado.utils import process
+from avocado import main
+
+
+class Avago9361(Test):
+
+    """
+    This class contains functions for VD operations
+    """
+    def setUp(self):
+
+        """
+        All basic set up is done here
+        """
+        self.controller = int(self.params.get('controller', default='0'))
+        self.disk = str(self.params.get('disk')).split(" ")
+        self.raid_level = str(self.params.get('raid_level', default='0'))
+        self.size = str(self.params.get('size', default='all'))
+        if not self.disk:
+            self.skip("Please provide disk to perform VD operations")
+        self.number_of_disk = len(self.disk)
+
+        self.dict_raid = {'r0': [1, None, None], 'r1': [2, None, 'Even'],
+                          'r5': [3, None, None], 'r6': [3, None, None],
+                          'r00': [4, None, 'Even'], 'r10': [4, None, 'Even'],
+                          'r50': [6, None, 'Multiple3'],
+                          'r60': [6, None, 'Multiple3']}
+        self.value = self.dict_raid[self.raid_level]
+
+        if self.number_of_disk < self.value[0]:
+            self.skip("Please give enough number of drives to create %s"
+                      % self.raid_level)
+        if self.value[2] == 'Multiple3':
+            if self.number_of_disk % 3 != 0:
+                if self.number_of_disk / 3 == 1:
+                    self.disk = self.disk[:-1]
+                else:
+                    self.disk = self.disk[:-2]
+
+        if self.value[2] == 'Even':
+            if self.number_of_disk % 2 != 0:
+                self.disk = self.disk[:-1]
+        self.raid_disk = ",".join(self.disk).strip(" ")
+        if self.raid_level == 'r10' or self.raid_level == 'r00':
+            self.pdperarray = 2
+        elif self.raid_level == 'r50' or self.raid_level == 'r60':
+            self.pdperarray = 3
```

Comment:
`self.value[1]` is never used. Either use it for max disks check, or eliminate it from the dictionary